### ISSUE TITLE
rx.select width applies incorrectly when stacked horizontally

### ISSUE THREAD

**Issue by hjpr** [reflex-dev/reflex#2824]
if applying a width of 100% to a single rx.select, the select component appropriately fills the container it resides in. if two rx.select boxes are beside each other and have width of 100%, instead of each select taking up the available space that they can it applies the full width of the container to both thus overfilling the container by double.

the workaround is apply a width of 50% to both, but is inconsistent with CSS usage.

```
def dropdowns():
    return rx.flex(
        rx.select(
            width='100%'
        ),
        rx.select(
            width='100%'
        )
    )
```

Version: Reflex 0.4.x

**Comment by hjpr** [comment]
Performing this in CSS means that both components share 100% of the available space thus functionally making each component take up half of the space even though their width is set to 100%. A behavior that all other reflex components follow (so far that I've seen).

For a given container of 200px, currently each select is 200px wide overflowing the container. Expected behavior in CSS is that each select should be 100px wide even though width is set to 100%.

**Comment by picklelo** [comment]
I think this is a bug - `select` shouldn't have a `width` prop. I'll put up a PR to remove this, thanks for reporting

**Comment by masenf** [comment]
I think we need to revert the fix in #2835 because it makes it impossible to set the width with the high level select.

the problem seems to be that radix buttons and select trigger set `flex_shrink: 0` CSS prop that prevents the element from shrinking to fit when the size is overspecified.

> The width of both of the above containers is set to 100%. They adhere to the max-width of the parent container of 768px. No overlap. No overflow.

If those child containers have `flex_shrink="0"` set, then they will exhibit this weirdness as well.

So I think ultimately CSS rules are being followed, there are just some "extra" rules introduced by radix themes that may be unexpected.
